### ISSUE TITLE
[TODO] fix:some weights are not trained

### ISSUE THREAD

**Issue by L-M-Sherlock** [open-spaced-repetition/fsrs-optimizer-burn#2]
Initial weights:

```rust
[
                0.4, 0.6, 2.4, 5.8, // initial stability
                4.93, 0.94, 0.86, 0.01, // difficulty
                1.49, 0.14, 0.94, // success
                2.18, 0.05, 0.34, 1.26, // failure
                0.29, 2.61, // hard penalty, easy bonus
            ]
```

Trained weights:
```
[src/training.rs:122] &model_trained.w.clone() = Param {
    id: ParamId {
        value: "f8c5b41d-186f-4b76-9760-c166e3e0df97",
    },
    value: Tensor {
        primitive: ADTensor {
            primitive: NdArrayTensor {
                array: [0.524094, 0.9928922, 2.9164228, 5.49887, 4.93, 0.94, 0.86, 0.01, 1.49, 0.14, 0.94, 2.18, 0.05, 0.34, 1.26, 0.29, 2.61], shape=[17], strides=[1], layout=CFcf (0xf), dynamic ndim=1,
            },
            node: Node {
                parents: [],
                order: 0,
                id: NodeID {
                    value: "31dc7e00-851a-4c20-aa78-dd1f8d1fae2a",
                },
                requirement: Grad,
            },
            graph: Graph {
                steps: Mutex { data: {
                    NodeID {
                        value: "31dc7e00-851a-4c20-aa78-dd1f8d1fae2a",
                    }: RootStep {
                        node: Node {
                            parents: [],
                            order: 0,
                            id: NodeID {
                                value: "31dc7e00-851a-4c20-aa78-dd1f8d1fae2a",
                            },
                            requirement: Grad,
                        },
                    },
                }},
            },
        },
    },
}
```

Some weights are not trained. Need to check the gradients during the optimization.

**Comment by L-M-Sherlock** [comment]
Closed by b751177d0b893cb7372f233d31634608986826ab

**Comment by L-M-Sherlock** [comment]
```
[src/training.rs:124] &model_trained.w.clone() = Param {
    id: ParamId {
        value: "9ad5f781-2856-4302-aba0-4f87b36650a8",
    },
    value: Tensor {
        primitive: ADTensor {
            primitive: NdArrayTensor {
                array: [0.2259229, 0.6, 1.9517553, 5.0650554, 5.4109883, 0.997119, 1.1177697, 0.01, 0.86084074, 0.55591977, 0.2646879, 1.9531487, 0.27267408, 0.086273395, 1.0594337, -0.00071644306, 2.4137843], shape=[17], strides=[1], layout=CFcf (0xf), dynamic ndim=1,
            },
            node: Node {
                parents: [],
                order: 0,
                id: NodeID {
                    value: "c23d485c-93af-400a-a9d2-5770dfc6a9a8",
                },
                requirement: Grad,
            },
            graph: Graph {
                steps: Mutex { data: {
                    NodeID {
                        value: "c23d485c-93af-400a-a9d2-5770dfc6a9a8",
                    }: RootStep {
                        node: Node {
                            parents: [],
                            order: 0,
                            id: NodeID {
                                value: "c23d485c-93af-400a-a9d2-5770dfc6a9a8",
                            },
                            requirement: Grad,
                        },
                    },
                }},
            },
        },
    },
}
```
